scanpy/tools/_leiden.py

```diff
@@ -134,22 +138,40 @@ def leiden(
             adjacency=adjacency,
         )
     # convert it to igraph
+    if not flavor == "leidenalg" and directed:
+        raise ValueError(
+            "Cannot use igraph's leiden implemntation with a directed graph."
+        )
     g = _utils.get_igraph_from_adjacency(adjacency, directed=directed)
     # flip to the default partition type if not overriden by the user
-    if partition_type is None:
+    if partition_type is None and flavor == "leidenalg":
         partition_type = leidenalg.RBConfigurationVertexPartition
+    elif not flavor == "leidenalg" and partition_type is not None:
+        raise ValueError("Do not pass in partition_type argument when using igraph.")
     # Prepare find_partition arguments as a dictionary,
     # appending to whatever the user provided. It needs to be this way
     # as this allows for the accounting of a None resolution
     # (in the case of a partition variant that doesn't take it on input)
     if use_weights:
-        partition_kwargs["weights"] = np.array(g.es["weight"]).astype(np.float64)
-    partition_kwargs["n_iterations"] = n_iterations
-    partition_kwargs["seed"] = random_state
+        clustering_args["weights"] = (
+            "weight"
+            if not flavor == "leidenalg"
+            else np.array(g.es["weight"]).astype(np.float64)
+        )
+    clustering_args["n_iterations"] = n_iterations
+    if flavor == "leidenalg":
+        clustering_args["seed"] = random_state
     if resolution is not None:
-        partition_kwargs["resolution_parameter"] = resolution
+        clustering_args["resolution_parameter"] = resolution
     # clustering proper
-    part = leidenalg.find_partition(g, partition_type, **partition_kwargs)
+    if not flavor == "leidenalg":
+        objective_function = clustering_args.pop("objective_function", "modularity")
+        with _utils.set_igraph_random_state(random_state):
+            part = g.community_leiden(
+                objective_function=objective_function, **clustering_args
+            )
```

Comment:
This would work too, no?

```suggestion
        clustering_args.setdefault("objective_function", "modularity")
        with _utils.set_igraph_random_state(random_state):
            part = g.community_leiden(**clustering_args)
```